src/main/java/seedu/address/logic/parser/AssignCommandParser.java

```diff
@@ -64,12 +65,18 @@ public AssignCommand parse(String arg) throws ParseException {
         Date date = ParserUtil.parseDate(argMultimap.getValue(PREFIX_DATE).get());
         Time time = ParserUtil.parseTime(argMultimap.getValue(PREFIX_TIME).get());
 
-        // Expired date time can only be confirmed when date and time are put together.
+        // expired date time can only be confirmed when date and time are put together.
         LocalTime timeWithLeeway = time.getTime().plusMinutes(CREATION_OFFSET_MINUTES);
         if (DateTimeUtil.isExpired(date.getDate(), timeWithLeeway)) {
             throw new ParseException(MESSAGE_EXPIRED_DATE_TIME);
         }
 
+        // check if the duration exceeds the CLOSING_TIME
+        Duration maxDuration = Duration.between(time.getTime(), CLOSING_TIME);
+        if (duration.compareTo(maxDuration) > 0) {
+            throw new ParseException(Time.MESSAGE_CONSTRAINTS);
+        }
```

Comment:
Maybe it would be better if the check is done in `ParserUtil.parseDuration` instead? This way the check will apply to `ChangeCommandParser` too. (Can just compare the duration with `CLOSING_TIME - OPENING_TIME`)

Lemme know what you think about it?

Reply:
if using the `CLOSING_TIME - OPENING_TIME`, there will not be a `ParserException` if the user enters something smaller than the limit but still out of operation hour, since this will only give an error for ending time is before start time in the appointment that cannot be viewed by the user.

Reply:
Hmm, true. Then I guess this will work 👍

Reply:
so I just keep it and change the `ChangCommand` also?